python/pyspark/sql/session.py

```diff
@@ -304,8 +329,13 @@ def __init__(
                 and not self._jvm.SparkSession.getDefaultSession().get().sparkContext().isStopped()
             ):
                 jsparkSession = self._jvm.SparkSession.getDefaultSession().get()
+                if options is not None:
+                    self.applyModifiableSetting(jsparkSession, options)
             else:
                 jsparkSession = self._jvm.SparkSession(self._jsc.sc(), options)
+        else:
+            if options is not None:
+                self.applyModifiableSetting(jsparkSession, options)
```

Comment:
Hm, in this case, there's an existing `jsparkSession`.  Do we still need to check the configurations?

Reply:
we apply options when there is a existed `SparkSession`